clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/codecs/OperationCodecFactory.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.internal.store.operations.codecs;
+
+import org.ehcache.spi.serialization.Serializer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationCodecFactory<K, V> {
+
+  private final Serializer<K> keySerializer;
+  private final Serializer<V> valueSerializer;
+
+  private final Map<OperationCode, OperationCodec<K>> operationCodeVsCodecMap;
+
+  public OperationCodecFactory(final Serializer<K> keySerializer, final Serializer<V> valueSerializer) {
+    this.keySerializer = keySerializer;
+    this.valueSerializer = valueSerializer;
+    operationCodeVsCodecMap = populateCodecMap();
+  }
+
+  private Map<OperationCode, OperationCodec<K>> populateCodecMap() {
+    Map<OperationCode, OperationCodec<K>> map = new HashMap<OperationCode, OperationCodec<K>>();
+    for (OperationCode code : OperationCode.values()) {
+      operationCodeVsCodecMap.put(code, getNewCodec(code));
+    }
+    return map;
+  }
+
+  public OperationCodec<K> getOperationCodec(OperationCode code) {
+    return operationCodeVsCodecMap.get(code);
+  }
+
+  private OperationCodec<K> getNewCodec(OperationCode code) {
```

Comment:
This could still be improved by having the enum return an `OperationCodecFactory` which would return the proper `OperationCodec` through a method taking the different `Serializer`s.